api/debug/flags.go

```diff
@@ -136,33 +187,106 @@ func CreateLogDir(logDir string) {
+		logger.Warn("The flag --debug is deprecated and will be removed in the future (v1.12.0), please use --log.debug")
+	}
+	log.PrintOrigins(debug)
 	log.ChangeGlobalLogLevel(glogger, log.Lvl(ctx.GlobalInt(verbosityFlag.Name)))
 	glogger.Vmodule(ctx.GlobalString(vmoduleFlag.Name))
-	glogger.BacktraceAt(ctx.GlobalString(backtraceAtFlag.Name))
+
+	backtrace := backtraceAtFlag.Value
+	if b := ctx.GlobalString(legacyBacktraceAtFlag.Name); b != "" {
+		backtrace = b
+		logger.Warn("The flag --backtrace is deprecated and will be removed in the future (v1.12.0), please use --log.backtrace")
+	}
+	if b := ctx.GlobalString(backtraceAtFlag.Name); b != "" {
+		backtrace = b
+	}
+	glogger.BacktraceAt(backtrace)
 	log.Root().SetHandler(glogger)
 
 	// profiling, tracing
-	runtime.MemProfileRate = ctx.GlobalInt(memprofilerateFlag.Name)
-	Handler.SetBlockProfileRate(ctx.GlobalInt(blockprofilerateFlag.Name))
+	runtime.MemProfileRate = memprofilerateFlag.Value
+	if ctx.GlobalIsSet(legacyMemprofilerateFlag.Name) {
+		runtime.MemProfileRate = ctx.GlobalInt(legacyMemprofilerateFlag.Name)
+		logger.Warn("The flag --memprofilerate is deprecated and will be removed in the future (v1.12.0), please use --pprof.memprofilerate")
+	}
+	if ctx.GlobalIsSet(memprofilerateFlag.Name) {
+		runtime.MemProfileRate = ctx.GlobalInt(memprofilerateFlag.Name)
+	}
+
+	blockProfileRate := blockprofilerateFlag.Value
+	if ctx.GlobalIsSet(legacyBlockprofilerateFlag.Name) {
+		blockProfileRate = ctx.GlobalInt(legacyBlockprofilerateFlag.Name)
+		logger.Warn("The flag --blockprofilerate is deprecated and will be removed in the future (v1.12.0), please use --pprof.blockprofilerate")
+	}
+	if ctx.GlobalIsSet(blockprofilerateFlag.Name) {
+		blockProfileRate = ctx.GlobalInt(blockprofilerateFlag.Name)
+	}
+	Handler.SetBlockProfileRate(blockProfileRate)
+
 	if traceFile := ctx.GlobalString(traceFlag.Name); traceFile != "" {
 		if err := Handler.StartGoTrace(traceFile); err != nil {
 			return err
 		}
 	}
-	if cpuFile := ctx.GlobalString(cpuprofileFlag.Name); cpuFile != "" {
+	cpuFileName := cpuprofileFlag.Name
+	if ctx.GlobalIsSet(legacyCpuprofileFlag.Name) {
+		cpuFileName = legacyCpuprofileFlag.Name
+		logger.Warn("The flag --cpuprofile is deprecated and will be removed in the future (v1.12.0), please use --pprof.cpuprofile")
+	}
+	if cpuFile := ctx.GlobalString(cpuFileName); cpuFile != "" {
 		if err := Handler.StartCPUProfile(cpuFile); err != nil {
 			return err
 		}
 	}
-	Handler.memFile = ctx.GlobalString(memprofileFlag.Name)
+	memProfile := memprofileFlag.Value
+	if ctx.GlobalIsSet(legacyMemprofileFlag.Name) {
+		memProfile = ctx.GlobalString(legacyMemprofileFlag.Name)
+		logger.Warn("The flag --memprofile is deprecated and will be removed in the future (v1.12.0), please use --pprof.memprofile")
+	}
+	if ctx.GlobalIsSet(memprofileFlag.Name) {
+		memProfile = ctx.GlobalString(memprofileFlag.Name)
+	}
+	Handler.memFile = ctx.GlobalString(memProfile)
```

Comment:
It looks having a pattern, so how about making a function? like checkLegacy or something else?

Reply:
@hyochan-brown I agree with jk. It would be much simple if function is provided. What do you think?

Reply:
I added it. Can you check if it's naming is okay?

Reply:
@hyochan-brown How about returning only flag name? Then I think only a function is needed.
```
func getFlagName(flag, legacyFlag cli.Flag, deprecateVersion string) string {
	if ctx.GlobalIsSet(flag.Name) {
		return flag.Name
	}
	if ctx.GlobalIsSet(legacyFlag.Name) {
		logger.Warn("The flag" + legacyFlag.Name + " is deprecated and will be removed in the future " + deprecateVersion + ",, please use " + flag.Name)
		return legacyFlag.Name
	}
}
```

Reply:
Sorry to late. 
I reflected it in this PR and Other flag PR.
Thanks JK!